x/datapool/types/message_pool.go

```diff
@@ -61,6 +79,49 @@ func (msg *MsgCreatePool) ValidateBasic() error {
 	if err != nil {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid curator address (%s)", err)
 	}
+
+	poolParams := msg.PoolParams
+
+	if len(poolParams.DataSchema) == 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "no data schema")
+	}
+
+	if poolParams.TargetNumData <= 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "target number of data should be greater than 0")
+	}
+
+	if poolParams.MaxNftSupply <= 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "maximum supply of NFT should be greater than 0")
+	}
+
+	if poolParams.NftPrice == nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "the price of NFT is empty")
+	}
+	if !poolParams.NftPrice.IsValid() {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "the price of NFT is invalid")
+	}
+
+	for _, validatorAddr := range poolParams.TrustedDataValidators {
+		_, err = sdk.AccAddressFromBech32(validatorAddr)
+		if err != nil {
+			return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid data validator address of %s", validatorAddr)
+		}
+	}
+
+	for _, issuerAddr := range poolParams.TrustedDataIssuers {
+		_, err = sdk.AccAddressFromBech32(issuerAddr)
+		if err != nil {
+			return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid issuer address of %s", issuerAddr)
+		}
+	}
+
+	if poolParams.Deposit == nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "the deposit is empty")
+	}
+	if !poolParams.Deposit.IsValid() {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "the deposit is invalid")
+	}
```

Comment:
In next PRs, we should define some genesis params (in the `genesis.json`) for this module: https://github.com/cosmos/cosmos-sdk/blob/master/x/slashing/types/params.go.

Reply:
I added TODO comment and will do it next PR. Thanks